.github/workflows/linux-build.yml

```diff
@@ -44,15 +40,71 @@ on:
       - "scripts/setup-helper-functions.sh"
       - ".github/workflows/linux-build.yml"
       - ".github/workflows/linux-build-base.yml"
-
 permissions:
   contents: read
-
 concurrency:
   group: ${{ github.workflow }}-${{ github.repository }}-${{ github.head_ref || github.sha }}
-  cancel-in-progress: true
-
+  cancel-in-progress: ${{github.ref != 'refs/heads/presto-0.293-clp-connector'}}
 jobs:
   linux-gcc:
     name: Build with GCC
-    uses: ./.github/workflows/linux-build-base.yml
+    runs-on: yscope-gh-runner
+    env:
+      CCACHE_DIR: "${{ github.workspace }}/ccache"
+      USE_CLANG: "${{ 'false' }}"
+    defaults:
+      run:
+        shell: bash
+        working-directory: velox
+    steps:
+      - name: Node Setup and Cleanup
+        working-directory: ${{ github.workspace }}
+        run: |
+          df -h
+          rm -rf /usr/local/lib/android
+          rm -rf /usr/share/dotnet
+          rm -rf /usr/share/dotnet
+          rm -rf /opt/ghc
+          rm -rf /opt/hostedtoolcache/CodeQL
+          rm -rf "/usr/local/share/boost"
+          rm -rf "$AGENT_TOOLSDIRECTORY"
+          docker image prune --all --force
+          docker builder prune -a
+          df -h
+      - uses: actions/checkout@v4
+        with:
+          path: velox
+      - name: Ensure Stash Dirs Exists
+        working-directory: ${{ github.workspace }}
+        run: |
+          mkdir -p '${{ env.CCACHE_DIR }}'
+      - name: Fetch the origin to get presto-0.293-clp-connector branch
+        run: |
+          git fetch
+      - name: Install Core Dependencies
+        run: |
+          source scripts/setup-ubuntu.sh && install_apt_deps
+      - name: Install Docs Dependencies
+        run: |
+          sudo apt install -y --no-install-recommends pandoc
+          sudo apt install -y pipx
+          pipx install --force uv cmake==3.31
+          make python-venv
+          ~/.local/bin/uv pip install -r scripts/docs-requirements.txt 
```

Comment:
Can we simplify this and the next step? Presumably we don't need pandoc since we aren't building the pyvelox docs, right? Ideally, I would prefer if building the docs was in a separate workflow so that it isn't blocked on all the other steps, etc.